clicache/src/DistributedSystem.cpp

```diff
@@ -140,9 +140,9 @@ namespace Apache
 
         // this we are calling after all .NET initialization required in
         // each AppDomain
-        auto nativeptr = native::DistributedSystem::create(mg_name.CharPtr, cache->GetNative().get(),
+        auto nativeptr = native::DistributedSystem::create(mg_name.CharPtr,
                                                             config->GetNative());
-        nativeptr->connect();
+        nativeptr->connect(cache->GetNative().get());
```

Comment:
This was split because of circular constructor dependency between CacheImpl and DistributedSystem.  DistributedSystem uses CacheImpl for StatisticsManager.  It worked before when DistributedSystem took a Cache, but we can't do that anymore since Cache may be moved.